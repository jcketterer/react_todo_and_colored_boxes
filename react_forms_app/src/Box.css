.Box {
  margin: 3px;
}
