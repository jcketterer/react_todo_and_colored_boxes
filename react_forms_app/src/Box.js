import React from 'react'
import './Box.css'

const Box = ({ id, handleRemove, width = 5, height = 5, backgroundColor = 'black' }) => {
  const remove = () => handleRemove(id)
  return (
    <div>
      <div
        className="Box"
        style={{
          height: `${height}em`,
          width: `${width}em`,
          backgroundColor,
        }}
      />
      <button onClick={remove}>Remove your box!</button>
    </div>
  )
}

export default Box
