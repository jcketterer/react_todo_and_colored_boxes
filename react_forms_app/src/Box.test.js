import React from 'react'
import { render } from '@testing-library/react'
import Box from './Box'

it('should render without breaking', () => {
  render(<Box />)
})

it('matches snap', () => {
  const { asFragment } = render(<Box />)
  expect(asFragment()).toMatchSnapshot()
})
