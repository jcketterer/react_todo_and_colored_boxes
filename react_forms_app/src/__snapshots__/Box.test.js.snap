// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snap 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Box"
      style="height: 5em; width: 5em; background-color: black;"
    />
    <button>
      Remove your box!
    </button>
  </div>
</DocumentFragment>
`;
